src/Routes/Default/Default.tsx

```diff
@@ -7,28 +7,57 @@ import { notificationsAtom, useRemoveNotification } from '../../state/notificati
 import Header from '../../Components/Header/Header';
 import React, { useEffect } from 'react';
 import useCurrentUser from '../../hooks/useCurrentUser';
-import { LayoutTypes, getWidgetMapping } from '../../api/dashboard-templates';
+import { LayoutTypes, WidgetPermission, getWidgetMapping } from '../../api/dashboard-templates';
 import { widgetMappingAtom } from '../../state/widgetMappingAtom';
 import '../../App.scss';
 import Portal from '@redhat-cloud-services/frontend-components-notifications/Portal';
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
 
 const DefaultRoute = (props: { layoutType?: LayoutTypes }) => {
   const isLayoutLocked = useAtomValue(lockedLayoutAtom);
   const notifications = useAtomValue(notificationsAtom);
   const removeNotification = useRemoveNotification();
   const setWidgetMapping = useSetAtom(widgetMappingAtom);
   const { currentUser } = useCurrentUser();
+  const { visibilityFunctions } = useChrome();
+
+  const checkPermissions = async (permissions: WidgetPermission[]): Promise<boolean> => {
+    const permissionResults = await Promise.all(
+      permissions.map(async (permission) => {
+        const { method, args } = permission;
```

Comment:
Might want to wrap the inner map function with `try ... catch` and return false from the catch branch.

Reply:
Good idea! Added this in.